--- ibus-1.5.2/src/ibusfactory.c~	2013-04-17 03:31:01.000000000 +0200
+++ ibus-1.5.2/src/ibusfactory.c	2013-07-06 13:03:48.733442060 +0200
@@ -287,12 +287,10 @@
             g_free (object_path);
         }
         else {
-            gchar *error_message = g_strdup_printf ("Can not fond engine %s", engine_name);
             g_dbus_method_invocation_return_error (invocation,
                                                    G_DBUS_ERROR,
                                                    G_DBUS_ERROR_FAILED,
-                                                   error_message);
-            g_free (error_message);
+                                                   "Can not fond engine %s", engine_name);
         }
         return;
     }
