From c93407a1056a3dc062f1efb79104f0ae32352b0a Mon Sep 17 00:00:00 2001
From: fujiwarat <takao.fujiwara1@gmail.com>
Date: Sat, 6 Oct 2012 17:54:27 +0900
Subject: [PATCH] Fix not to switch engines by non-trigger keys.

---
 ui/gtk3/switcher.vala |    6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/ui/gtk3/switcher.vala b/ui/gtk3/switcher.vala
index 2587117..ab2040b 100644
--- a/ui/gtk3/switcher.vala
+++ b/ui/gtk3/switcher.vala
@@ -123,6 +123,7 @@ class Switcher : Gtk.Window {
         /* Let gtk recalculate the window size. */
         resize(1, 1);
 
+        m_result = 0;
         m_selected_engine = index;
         m_label.set_text(m_buttons[index].longname);
         m_buttons[index].grab_focus();
@@ -338,6 +339,11 @@ class Switcher : Gtk.Window {
                     break;
                 default:
                     debug("0x%04x", pe->keyval);
+                    if (m_loop != null) {
+                        m_loop.quit();
+                        m_loop = null;
+                    }
+                    retval = false;
                     break;
             }
         } while (false);
-- 
1.7.10.4

